cmd/river/main.go

```diff
@@ -69,66 +90,104 @@ before starting the client, and works until all jobs are finished.
 
+	// migrate-down and migrate-up share a set of options, so this is a way of
+	// plugging in all the right flags to both so options and docstrings stay
+	// consistent.
+	addMigrateFlags := func(cmd *cobra.Command, opts *migrateOpts) {
+		cmd.Flags().StringVar(&opts.DatabaseURL, "database-url", "", "URL of the database to migrate (should look like `postgres://...`")
+		cmd.Flags().BoolVar(&opts.DryRun, "dry-run", false, "print information on migrations, but don't apply them")
+		cmd.Flags().IntVar(&opts.MaxSteps, "max-steps", 1, "maximum number of steps to migrate")
+		cmd.Flags().BoolVar(&opts.ShowSQL, "show-sql", false, "show SQL of each migration")
+		cmd.Flags().IntVar(&opts.TargetVersion, "target-version", 0, "target version to migrate to (final state includes this version, but none after it)")
+		mustMarkFlagRequired(cmd, "database-url")
+	}
+
 	// migrate-down
 	{
-		var opts migrateDownOpts
+		var opts migrateOpts
 
 		cmd := &cobra.Command{
 			Use:   "migrate-down",
 			Short: "Run River schema down migrations",
-			Long: `
+			Long: strings.TrimSpace(`
 Run down migrations to reverse the River database schema changes.
 
 Defaults to running a single down migration. This behavior can be changed with
 --max-steps or --target-version.
-	`,
+
+SQL being run can be output using --show-sql, and executing real database
+operations can be prevented with --dry-run. Combine --show-sql and --dry-run to
+dump prospective migrations that would be applied to stdout.
+	`),
 			Run: func(cmd *cobra.Command, args []string) {
-				execHandlingError(func() (bool, error) { return migrateDown(ctx, &opts) })
+				execHandlingError(func() (bool, error) { return migrateDown(ctx, makeLogger(), os.Stdout, &opts) })
 			},
 		}
-		cmd.Flags().StringVar(&opts.DatabaseURL, "database-url", "", "URL of the database to migrate (should look like `postgres://...`")
-		cmd.Flags().IntVar(&opts.MaxSteps, "max-steps", 1, "maximum number of steps to migrate")
-		cmd.Flags().IntVar(&opts.TargetVersion, "target-version", 0, "target version to migrate to (final state includes this version, but none after it)")
-		mustMarkFlagRequired(cmd, "database-url")
+		addMigrateFlags(cmd, &opts)
+		rootCmd.AddCommand(cmd)
+	}
+
+	// migrate-get
+	{
+		var opts migrateGetOpts
+
+		cmd := &cobra.Command{
+			Use:   "migrate-get",
+			Short: "Get SQL for specific River migration",
+			Long: strings.TrimSpace(`
+Retrieve SQL for a single migration version. This command is aimed at cases
+where using River's internal migration framework isn't desirable by allowing
+migration SQL to be dumped for use elsewhere.
+
+Specify a single version with --version, and one of --down or --up.
+	`),
+			Run: func(cmd *cobra.Command, args []string) {
+				execHandlingError(func() (bool, error) { return migrateGet(ctx, makeLogger(), os.Stdout, &opts) })
+			},
+		}
+		cmd.Flags().BoolVar(&opts.Down, "down", false, "print down migration")
+		cmd.Flags().BoolVar(&opts.Up, "up", false, "print up migration")
```

Comment:
It's a little odd that this command takes `--down` and `--up` options, when the migration commands are `migrate-down` and `migrate-up`, but I tried to consider all the alternatives, and I think this is the least bad. Having two full commands for `migrate-get-down` and `migrate-get-up` feels too heavy handed and unwieldy.